system-server/system_server/jwt/system_jwt.py

```
@@ -0,0 +1,60 @@
+"""Interface for interacting with JWT's."""
+import jwt
+import logging
+from datetime import datetime, timezone, timedelta
+from dataclasses import dataclass
+from typing import Optional
+
+_log = logging.getLogger(__name__)
+
+_JWT_ALGORITHM = "HS512"
+
+
+@dataclass
+class Registrant:
+    """Uniquely identifies a registered agent.
+
+    The tuple of (subject,agent,agent_id) provides a unique identification
+    of a registered agent.
+    """
+
+    subject: str
+    agent: str
+    agent_id: str
+
+
+def create_jwt(
+    signing_key: str,
+    duration: timedelta,
+    registrant: Registrant,
+    audience: str,
+    now: Optional[datetime] = None,
+) -> str:
+    """Generate a signed JWT with the specified parameters."""
+    if now is None:
+        now = datetime.now(tz=timezone.utc)
+    claims = {
+        "iat": now,
+        "exp": now + duration,
+        "sub": registrant.subject,
+        "ot_agent": registrant.agent,
+        "ot_aid": registrant.agent_id,
+        "aud": audience,
+    }
+    try:
+        return jwt.encode(payload=claims, key=signing_key, algorithm=_JWT_ALGORITHM)
+    except Exception as e:
+        _log.error(f"Error during JWT creation: {type(e)}:{e}")
+        raise e
```

Do we need to explicitly catch this exception and log it here?

If `jwt.encode()` raises, I'd expect the exception to bubble all the way up until it hits some top-level catch-all exception logger.